### PR TITLE
Add an option for `no-unresolved` rule that allows one to resolve .ts files from .js imports

### PR DESCRIPTION
**Pull request by ukstv**
One problem that prevents this package from being useable on TS codebase, is that it treats a file extension literally. TypeScript-based ES modules require you to import TS files using .js extension:
```
import * as foo from './foo.js'
```
The statement above really imports content of `./foo.ts` file.

This PR adds a new boolean option `ts`, set to `false` by default. If it is set to `true`, a statement above in a TS codebase is considered valid. If the plugin can not resolve `./foo.js`, it tries to also resolve `./foo.ts`, and then goes on as before.

### CODE REVIEW

**Comment by ntucker** [comment]
probably want to have it try .tsx as well

**Comment by ukstv** [comment]
Okay, now it tries to resolve `.jsx?` to `.ts` or `.tsx`.

**Review by Maxim-Mazurok — CHANGES_REQUESTED** [review]
I think that some tests for this need to be added to https://github.com/import-js/eslint-plugin-import/blob/main/tests/src/rules/no-unresolved.js

**Comment by ukstv** [comment]
@Maxim-Mazurok Added the tests. What is left for the PR to be merged?

**Comment by Maxim-Mazurok** [comment]
Thanks! I think the only thing left is to get some maintainer to approve it. I didn't really check the code, but now that we have tests I think that maintainers approval is more likely to happen.

**Comment by ukstv** [comment]
Good. Let me properly rebase that, and it is ready to merge then.

**Comment by ukstv** [comment]
Rebased now. Hope it is good to merge. Feel free to add comments/critique, so that I can amend the PR.

**Comment by ljharb** [comment]
Since this deals with TS-specific resolution, is there a reason this couldn't/shouldn't be an option on the TS eslint resolver? (cc @bradzacher)

**Comment by ukstv** [comment]
Should it be a default there, or hidden behind a flag, like here?

**Comment by bradzacher** [comment]
@ukstv consider raising an issue there to discuss with the maintainers! You may find there is already one?

**Comment by JounQin** [comment]
I don't understand, https://github.com/import-js/eslint-import-resolver-typescript already supports `exports` field in `package.json`, this is an unnecessary change in the core module.

**Comment by ukstv** [comment]
> I don't understand, https://github.com/import-js/eslint-import-resolver-typescript already supports `exports` field in `package.json`, this is an unnecessary change in the core module.

`exports` clause in `package.json` has nothing to do with this change. What we are trying to achieve is having a check if a local `./foo.js` exists because of corresponding `./foo.ts` exists.

**Comment by JounQin** [comment]
> What we are trying to achieve is having a check if a local ./foo.js exists because of corresponding ./foo.ts exists.

It is also supported by https://github.com/import-js/eslint-import-resolver-typescript

**Comment by JounQin** [comment]
As above, please use https://github.com/import-js/eslint-import-resolver-typescript instead.

**Comment by ntucker** [comment]
I cannot find the equivalent of "import/order" in [eslint-import-resolver-typescript](https://github.com/import-js/eslint-import-resolver-typescript).

**Comment by ljharb** [comment]
How is that unsafe?

**Comment by JounQin** [comment]
> [import-js/eslint-plugin-import#importextensions](https://github.com/import-js/eslint-plugin-import#importextensions) the docs probably need to be updated to recommend installing this instead of adding extensions which is unsafe?

I'm going to PR to add a section about TypeScript usage in README, if you want to do that, feel free to tell me and take it.

**Pull request closed without merging**
